org/mozilla/jss/util/ECCurve.java

```diff
@@ -0,0 +1,199 @@
+        new BigInteger("6864797660130609714981900799081393217269435300143305409394463459185543183397655394245057746333217197532963996371363321113864768612440380340372808892707005449"),
+        1 // OpenJDK lists this curve as having a cofactor of one, but NIST
+          // doesn't share their cofactor.
+    );
+
+    // OIDs are defined at the bottom of this class.
+
+    private String[] names;
+    private OBJECT_IDENTIFIER[] oids;
+
+    private ECField field;
+    private BigInteger a;
+    private BigInteger b;
+    private BigInteger generatorX;
+    private BigInteger generatorY;
+    private BigInteger order;
+    private int cofactor;
+
+    private ECCurve(String[] names, OBJECT_IDENTIFIER[] oids, ECField field,
+                    BigInteger a, BigInteger b, ECPoint generator,
+                    BigInteger order, int cofactor)
+    {
+        this.names = names;
+        this.oids = oids;
+        this.field = field;
+        this.a = a;
+        this.b = b;
+        this.generatorX = generator.getAffineX();
+        this.generatorY = generator.getAffineY();
+        this.order = order;
+        this.cofactor = cofactor;
+    }
+
+    private ECCurve(String[] names, OBJECT_IDENTIFIER[] oids, ECField field,
+                    BigInteger a, BigInteger b, BigInteger generatorX,
+                    BigInteger generatorY, BigInteger order, int cofactor)
+    {
+        this.names = names;
+        this.oids = oids;
+        this.field = field;
+        this.a = a;
+        this.b = b;
+        this.generatorX = generatorX;
+        this.generatorY = generatorY;
+        this.order = order;
+        this.cofactor = cofactor;
+    }
+
+    public String[] getNames() {
+        return this.names;
+    }
+
+    public OBJECT_IDENTIFIER[] getOIDs() {
+        return this.oids;
+    }
+
+    public ECField getField() {
+        return field;
+    }
+
+    public BigInteger getA() {
+        return a;
+    }
+
+    public BigInteger getB() {
+        return b;
+    }
+
+    public EllipticCurve getEC() {
+        return new EllipticCurve(field, a, b);
+    }
+
+    public BigInteger getGeneratorX() {
+        return generatorX;
+    }
+
+    public BigInteger getGeneratorY() {
+        return generatorY;
+    }
+
+    public ECPoint getGenerator() {
+        return new ECPoint(generatorX, generatorY);
+    }
+
+    public BigInteger getOrder() {
+        return order;
+    }
+
+    public ECParameterSpec getECParameterSpec() {
+        return new ECParameterSpec(getEC(), getGenerator(), getOrder(), getCofactor());
+    }
+
+    public int getCofactor() {
+      return cofactor;
+    }
+
+    public static ECCurve fromOID(OBJECT_IDENTIFIER oid) {
+        for (ECCurve curve : ECCurve.values()) {
+            for (OBJECT_IDENTIFIER curve_oid : curve.oids) {
+                if (curve_oid.equals(oid)) {
+                    return curve;
+                }
+            }
+        }
+
+        return null;
+    }
+
+    // OID Base Arcs
```

Comment:
@edewata how should I store these? If I introduce them in the `jss.util.EC` class, they'll be publicly accessible, which I guess is fine. Perhaps I should make a new `jss.util.ECOIDs` class for storing them? I can't use them in this enum because Java complains about forward references. I think it'd make the definition up top a lot easier and a lot less error prone. We can also deduplicate them out of `PKC11KeyPairGenerator` once we get a final resting place for them.

Thoughts?

Reply:
I've introduced an ECOIDs class which I'll use for this.